### ISSUE TITLE
read multiple Lance datasets directly.

### ISSUE THREAD

**Issue by jianlong-yuan** [lancedb/lance#5051]
I hope to load multiple .lance datasets directly like this.
Is there a method or recommended way to achieve this?

```
import lance

lance_paths = [
    "**/data1.lance",
    "**/data2.lance",
]

ds = lance.dataset(lance_paths)
```

**Comment by westonpace** [comment]
Are these two different tables that you want to join?  Or have you partitioned your data into two tables with identical schemas but two sets of rows?

**Comment by jianlong-yuan** [comment]
> Are these two different tables that you want to join? Or have you partitioned your data into two tables with identical schemas but two sets of rows?

My goal is to read data from both tables at the same time, allowing random sampling from either or both of them during PyTorch training.

**Comment by jianlong-yuan** [comment]
any news?

**Comment by westonpace** [comment]
Does a pytorch [`ConcatDataset`](https://docs.pytorch.org/docs/stable/data.html#torch.utils.data.ConcatDataset) work for you?  We'll probably add something like this of our own pretty soon once https://github.com/lancedb/lancedb/pull/2725 lands

**Comment by Xuanwo** [comment]
Closing this support/API request as not planned because the discussion already identified composition through PyTorch datasets, while a native multi-dataset API was never defined for schema compatibility, sampling ratios, shuffle, workers, or distributed semantics. @jianlong-yuan, please reopen with that explicit contract if composition is insufficient.
